enhancements/cloud-integration/tokenized_auth_enablement_operators_on_cloud.md

```diff
@@ -0,0 +1,527 @@
+Operators that must reach the AWS API, should be enabled to work with STS credentials in a consistent, and automated 
+fashion that allows customer to use those operators as easily as possible, driving the use of layered products.
+
+As an operator team that may already be supporting cloud credential provisioning, including a manner for allowing
+operand instances to have a unique set of credential per instance, I want my solution to keep working (Quay Operator).
+
+
+### Goals
+
+Allow Day-2 Operators to access cloud provider resources as seamlessly as possible when being installed, used and
+updated on STS enabled clusters.
+
+While providing the above goal, allow for multi-tenancy. In this sense multi-tenancy means that an operator may
+provide different cloud provider credentials per operand such that operator users may uniquely access non-shared cloud
+resources.
+
+Guide and assist Day-2 Operator admins in providing the required cloud provider credentials matched to their permission
+needs for install and update.
+
+Ideally, a solution here will work in both HyperShift (STS always) and non-HyperShift clusters.
+
+### Non-Goals
+
+Day 1 operators are not included in this proposal, they are pre-provisioned for STS enabled clusters (at least 
+for HyperShift)
+
+
+
+## Proposal
+
+[NB: these are all short now, just adding to them as we outline what each involved component will do in the solution]
+
+Cloud Credential Operator (CCO) changes: Add an STS mode, distinct from disabled/manual mode that will look for and
+process CredentialRequests referenced in Operator CRs. This will operate in one of two ways depending on which other parts 
+of this EP for other components are adopted: specifically HyperShift/Rosa adding CCO with STS mode or continuing with the
+current support for the pod identity webhook).
+- CCO STS mode flow will work by adding a "role-arn"-type field on operator added CredentialsRequest objects, this will 
+trigger CCO to annotate the ServiceAccount on behalf of the operator.
+- Webhook mode will work by annotating the ServiceAccount triggering the projection of the service account tokens into pods
+ created for the operator (same method and resultant actions as: https://github.com/openshift/hypershift/pull/1351)
+
+HyperShift changes: Possibly some changes to logic for Pod Identity Webhooks as needed for fine-grained credential
+management to allow for multi-tenancy (see definition earlier). Also, need to add pod identity webhooks targeting other
+cloud providers and change the annotations to more generic naming.
+
+OperatorHub changes: Allow for import of additional, well-known, ENV variables on the Subscription object. These fields
+will allows UX for the information needed by CCO or webhook, for input of the cloud credentials. 
+
+OLM changes: Generate a manual ack for operator upgrade when cloud resources are involved. This can be determined by 
```

Comment:
I think we punted on this one for now as it's just outside the domain of OLM to handle or manage cloud resource requests. Instead we should enforce three things:
1. all operators should follow semver rules and shipping a z-release that requires extra permissions should not be allowed
2. even if you need extra permissions, from one minor to the next (at least) that shouldn't break existing operations - only new ones that require that elevated set of perms
3. operators should have channels per z-stream and customers should understand that manually changing the subscription to a different channel is effectively an "ack". When FBC lands, we'll have more options for update graphs.